csharp_dotnetcore/3.Welcome-Users/WelcomeUserBot.cs

```diff
@@ -0,0 +1,146 @@
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Newtonsoft.Json;
+using WelcomeUser.State;
+
+namespace WelcomeUser
+{
+    /// <summary>
+    /// Represents a bot that can process incoming activities.
+    /// For each interaction from the user, an instance of this class is called.
+    /// For each Activity received, a new instance of this class is created.
+    /// Objects that are expensive to construct, or have a lifetime.
+    /// beyond the single Turn, should be carefully managed and chached - see Startup.cs.
+    /// </summary>
+    public class WelcomeUserBot : IBot
+    {
+        // generic message sent to user
+        private const string _genericMessage = "This is a simple Welcome Bot sample.You can say \'intro\' to see the introduction card. If you are running this bot in the Bot Framework emulator, press the \'Start Over\' button to simulate user joining a bot or a channel";
+
+        // the bot state accessor object. Use this to access specific state properties
+        private readonly WelcomeUserStateAccessors _welcomeUserStateAccessors;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WelcomeUserBot"/> class.
+        /// Initalize WelcomeUserBot.
+        /// </summary>
+        /// <param name="statePropertyAccessor"> Bot state accessor object.</param>
+        public WelcomeUserBot(WelcomeUserStateAccessors statePropertyAccessor)
+        {
+            _welcomeUserStateAccessors = statePropertyAccessor ?? throw new System.ArgumentNullException("state accessor can't be null");
+        }
+
+        /// <summary>
+        /// Every Conversation turn for our WelcomeUser Bot will call this method, including
+        /// any type of activiaties such as ConversationUpdate or ContactRelationUpdate which
+        /// are sent when a user joins a conversation.
+        /// This bot doesn't use any dialogs; it's "single turn" processing, meaning a single
+        /// request and response, with no stateful conversation between user input.
+        /// This bot does use UserState to keep track of first message users send to the bot.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = new CancellationToken())
+        {
+            // use state accessor to extract the didBotWelcomeUser flag
+            var didBotWelcomeUser = await _welcomeUserStateAccessors.DidBotWelcomedUser.GetAsync(turnContext, () => false);
```

Comment:
Why the () => false lambda expression? I don't think it's needed. If it is needed, please add a comment as to what it's doing, as I honestly don't know. I *assume* it's the default value, but am not sure and believe that should be set in the POCO and not here.